ui/app/AppLayouts/Chat/ContactsColumn/ChannelList.qml

```diff
@@ -83,17 +83,66 @@ ScrollView {
 
         Separator {}
 
-        PopupMenu {
-            hasArrow: false
-            title: qsTr("Mute Chat")
-
-            // TODO implement mute chat in Model and call it here
-            Action { text: qsTr("15 minutes"); icon.width: 0; }
-            Action { text: qsTr("1 hour"); icon.width: 0; }
-            Action { text: qsTr("8 hours"); icon.width: 0; }
-            Action { text: qsTr("24 hours"); icon.width: 0; }
-            Action { text: qsTr("Until I turn it back on"); icon.width: 0; }
+        Action {
+            text: chatsModel.channelIsMuted(channelContextMenu.channelIndex) ? qsTr("Unmute chat") : qsTr("Mute chat")
+            icon.source: "../../../img/bell.svg"
+            icon.width: 16
+            icon.height: 16
+            onTriggered: {
+              if (chatsModel.channelIsMuted(channelContextMenu.channelIndex)) {
+                chatsModel.unmuteChannel(channelContextMenu.channelIndex)
+                return
+              }
+              chatsModel.muteChannel(channelContextMenu.channelIndex)
+            }
```

Comment:
There one tiny issue here: this needs to be rerendered when users mute/unmute channels otherwise the UI keeps showing e.g. "Mute chat" even though it has been muted already.

I thought I could trigger a rerender through `channelList.updateChat()` but that didn't seem to do it.. if anyone knows how we can trigger a re-render here, that'd be wonderful!

Reply:
I think the problem is that you get the `muted` status from a `View` function instead than from the Channel itself, so the `onChanged` signal doesn't propagate to you.

One option is to try to change the text from the QML. i.e. when you trigger the mute or unmute, just manually set `this.text = ...`. If on re-render, it re-fetches the `muted` status from the backend, then it's the easiest way.

Another way would be to try to add a `Connection` to the `ChannelList` for the `onChanged` signal, but I'm not an expert with that so it might be trickier.

Reply:
Okay I've exposed the `muted` property on the `Channel` and pass it back up to `openPopup` when the menu is opened. From there, reading the property does the trick!